File: Aurora/Source/AspireOverflow/Models/Query.cs
using AspireOverflow.Models.Interfaces;
namespace AspireOverflow.Models
{
    public class Query :IAuditField
    {
        public Query(){
            QueryComments = new HashSet<QueryComment>();
        }
        public int Id { get; set; }

        public string Title { get; set; }
        public string Content { get; set; }
        public string code { get; set; }

        public bool IsSolved { get; set; }

      
 
        public int CreatedBy { get; set; }

        public DateTime CreatedOn { get; set; }
        public int? UpdatedBy { get; set; }

        public DateTime? UpdatedOn { get; set; }
        


      public ICollection<QueryComment>? QueryComments {get;set;}
    }
}

File: Aurora/Source/Angular/Models/Query.ts
export class Query {
    queryId: number=0;
    title!: string;
    content!: string;
    code: string="";
    isSolved: boolean=false;
    queryComments:QueryComment[]=[];
    constructor(){}
    
}

export class QueryComment{
        queryCommentId: number=0;
        comment: string="";
        createdOn: string="";
        queryId: number=0;
        code: string="";
    
}